File: OpenWeather/app/src/main/java/com/example/openweather/Hilt/RepositoryModule.java
package com.example.openweather.Hilt;

import com.example.openweather.Hilt.Qualifier.LocalRepository;
import com.example.openweather.Hilt.Qualifier.RemoteRepository;
import com.example.openweather.Repository.LocalRepository.LocalRepositoryImpl;
import com.example.openweather.Repository.RemoteRepository.RemoteRepositoryImpl;
import com.example.openweather.Repository.Repository;

import java.lang.annotation.ElementType;
import java.lang.annotation.Target;

import javax.inject.Qualifier;

import dagger.Binds;
import dagger.Module;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;


@Module
@InstallIn(SingletonComponent.class)
public abstract class RepositoryModule {

    @Binds
    @RemoteRepository
    abstract Repository bindRemoteRepository(RemoteRepositoryImpl remoteRepository);

    @Binds
    @LocalRepository
    abstract Repository bindLocalRepository(LocalRepositoryImpl localRepository);

}

File: Recipes/app/src/main/java/com/example/recipes/Hilt/RepositoryModule.kt
package com.example.recipes.Hilt

import com.example.recipes.Model.Repository.RecipeRepository
import com.example.recipes.Model.Repository.RecipeRepositoryImpl
import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent

@Module
@InstallIn(SingletonComponent::class)
abstract class RepositoryModule {

    @Binds
    abstract fun bindRecipeRepository(recipeRepository : RecipeRepositoryImpl) : RecipeRepository
}